Remove txt2tags dependency (use markdown instead) (#36966)
qgis/QGIS: INSTALL.md

QGIS
Building QGIS from source - step by step

   1. [Introduction](#1-introduction)
   2. [Overview](#2-overview)
   3. [Building on GNU/Linux](#3-building-on-gnulinux)

>    4. [Building on Windows](#3-building-on-windows)

Reviewer: ```suggestion
   4. [Building on Windows](#4-building-on-windows)
```

>    5. [Building on MacOS X](#4-building-on-macos-x)

Reviewer: ```suggestion
   5. [Building on MacOS X](#5-building-on-macos-x)
```

  OSGeo4W[64]\include, that conflicts with Visual Studio own stdint.h, which for
  example breaks the build of the virtual layer provider.

Earlier versions of this document also covered how to build all above
dependencies.  If you're interested in that, check the history of this page in the Wiki
or the SVN repository.

ninja:
  copy ninja.exe to C:\OSGeo4W64\bin\

### 4.1.3. Clone the QGIS Source Code

Choose a directory to store the QGIS source code.
For example, to put it in the OSGeo4W64 install, navigate there:

```cmd
    cd C:\OSGeo4W64
```

This directory will be assumed for all instructions
that follow.

On the command prompt clone the QGIS source from
git to the source directory `QGIS`:

```cmd
    git clone git://github.com/qgis/QGIS.git
```

This requires Git. If you have Git for Windows on your PATH already,
you can do this from a normal command prompt. If you do not, you can
use the Git package that was installed as part of Cygwin by opening
a Cygwin[64] Terminal

And, to avoid Git in Windows reporting changes to files not actually modified:

```cmd
    cd QGIS
    git config core.filemode false
```

> ### 4.1.4. Setting up the Visual Studio project with CMake

Reviewer: ```suggestion
### 4.1.4. Configure and build with CMake from command line
```

installed in the default locations):

```cmd
    @echo off
    call C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\msvc-env.bat x86_64
    @cmd
```

Save the batch file as `C:\OSGeo4W64\OSGeo4W-dev.bat` and run it.

#### 4.1.4.1 Using configonly.bat to create the MSVC solution file:
  We will be using the file `ms-windows/osgeo4w/configonly.bat` to create an MSVC solution file.
  There are two supported CMake generators for creating a solution file: Ninja, and native MSVC.
  The advantage of using native MSVC solution is that you can find the root of build problems much more easily.
  configonly.bat is meant to create a configured build directory with a MSVC solution file:

```cmd
    cd C:\OSGeo4W64\QGIS\ms-windows\osgeo4w
    configonly.bat
```

#### 4.1.4.2 Compiling QGIS with MSVC:
  We will need to run MSVC with all the environment variables set, thus we will run it as follows:
  Run the batch file OSGeo4W-dev.bat you created before.
  On the command prompt run `call gdal-dev-env.bat` to add the release gdal and proj libraries to your PATH.
  On the command prompt run `devenv` to open MSVC.
  From MSVC, open the solution file

>   `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\qgis.sln`

Reviewer: ```suggestion
  `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\qgis.sln`.
```

  Try to build the solution (go grab a cup of tea, it may take a while).
  If it fails, run it again and again until there are (hopefully) no errors.

> Running QGIS from within MSVC:

Reviewer: I think a header section here would be worthwhile for consistency with 4.1.4.1 and 4.1.4.2

>   Edit the properties of the project ALL_BUILD:

Reviewer: ```suggestion
  Edit the properties of the project ALL_BUILD to include the path to the executable:
```

>   Debugging -> Command -> `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\output\bin\RelWithDebInfo\qgis.exe`

Reviewer: ```suggestion
  Debugging -> Command -> `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\output\bin\RelWithDebInfo\qgis.exe`.
```

>   Run: Debug -> Start Debugging (F5) or Start Without Debugging (Ctrl+F5)

Reviewer: ```suggestion
  To run, use the menu commands: Debug -> Start Debugging (F5) or Start Without Debugging (Ctrl+F5).
```

  bindings are placed into the output directory when you run the INSTALL target.

  Hit `Configure` to start the configuration and select `Visual Studio 9 2008`
  and keep `native compilers` and click `Finish`.

  The configuration should complete without any further questions and allow you to
  click `Generate`.

  Now close `cmake-gui` and continue on the command prompt by starting
  `vcexpress`.  Use File / Open / Project/Solutions and open the
  qgis-x.y.z.sln File in your project directory.

  Change `Solution Configuration` from `Debug` to `RelWithDebInfo` (Release
  with Debug Info)  or `Release` before you build QGIS using the ALL_BUILD
  target (otherwise you need debug libraries that are not included).

  After the build completed you should install QGIS using the INSTALL target.

  Install QGIS by building the INSTALL project. By default this will install to
  c:\Program Files\qgis<version> (this can be changed by changing the
  CMAKE_INSTALL_PREFIX variable in cmake-gui).

  You will also either need to add all the dependency DLLs to the QGIS install
  directory or add their respective directories to your PATH.

> ### 4.1.5. Packaging

Reviewer: ```suggestion
### 4.1.6. Packaging
```

To create a standalone installer there is a perl script named 'creatensis.pl'
in 'qgis/ms-windows/osgeo4w'.  It downloads all required packages from OSGeo4W
and repackages them into an installer using NSIS.

The script can be run on both Windows and Linux.

On Debian/Ubuntu you can just install the 'nsis' package.

NSIS for Windows can be downloaded at:

> http://nsis.sourceforge.net

And Perl for Windows (including other requirements like 'wget', 'unzip', 'tar'
and 'bzip2') is available at:

> http://cygwin.com

> ### 4.1.6. Packaging your own build of QGIS

Reviewer: ```suggestion
### 4.1.7. Packaging your own build of QGIS
```

Assuming you have completed the above packaging step, if you want to include
your own hand built QGIS executables, you need to copy them in from your
windows installation into the ms-windows file tree created by the creatensis
script.

```cmd
    cd ms-windows/
    rm -rf osgeo4w/unpacked/apps/qgis/*
    cp -r /tmp/qgis1.7.0/* osgeo4w/unpacked/apps/qgis/
```

Now create a package.

```cmd
    ./quickpackage.sh
```

After this you should now have a nsis installer containing your own build
of QGIS and all dependencies needed to run it on a windows machine.

> ### 4.1.7. Osgeo4w packaging

Reviewer: ```suggestion
### 4.1.8. Osgeo4w packaging
```
